Common/cpp/Tools/RuntimeDecorator.cpp

```diff
@@ -188,4 +182,83 @@ void RuntimeDecorator::decorateRNRuntime(
       rnRuntime, "_REANIMATED_IS_REDUCED_MOTION", isReducedMotion);
 }
 
+std::string parseValue(jsi::Runtime &rt, jsi::Value const &value) {
+  if (value.isBool() || value.isNumber()) {
+    return value.toString(rt).utf8(rt);
+  } else if (value.isString()) {
+    return "\"" + value.getString(rt).utf8(rt) + "\"";
+  } else if (value.isUndefined()) {
+    return "undefined";
+  } else if (value.isObject()) {
+    return parseComplexValue(rt, value.getObject(rt));
+  } else {
+    return "unsupported value type";
+  }
+}
+
+std::string parseComplexValue(jsi::Runtime &rt, jsi::Object const &object) {
+  std::stringstream parsed;
+
+  if (object.isArray(rt)) {
+    parsed << "[";
+
+    jsi::Array arr = object.getArray(rt);
+    size_t length = arr.size(rt);
+
+    for (size_t i = 0; i < length; i++) {
+      jsi::Value element = arr.getValueAtIndex(rt, i);
+      parsed << parseValue(rt, element) << ", ";
+    }
+
+    if (length > 0) {
+      parsed.seekp(-2, parsed.cur);
+    }
+    parsed << "] ";
+  } else if (object.isFunction(rt)) {
+    parsed << "[Function "
+           << object.getProperty(rt, "name").toString(rt).utf8(rt) << "]";
+  } else if (object.isHostObject(rt)) {
+    int status;
+    jsi::HostObject *hostObject = object.asHostObject(rt).get();
+    const char *hostObjClassName =
+        abi::__cxa_demangle(typeid(*hostObject).name(), NULL, NULL, &status);
+    if (status == 0) {
+      // create a plain jsi::Object
+      facebook::jsi::Object obj(rt);
+
+      // copy properties from hostObject to obj
+      for (auto &key : hostObject->getPropertyNames(rt)) {
+        facebook::jsi::Value value = hostObject->get(rt, key);
+        obj.setProperty(rt, key, value);
+      }
```

Comment:
Is it possible to do it without creating a new object and copying all properties?